internal/command/volumes/extend.go

```diff
@@ -72,7 +73,22 @@ func runExtend(ctx context.Context) error {
 		return err
 	}
 
-	sizeGB := flag.GetInt(ctx, "size")
+	sizeFlag := flag.GetString(ctx, "size")
+	sizeBytes, err := units.FromHumanSize(sizeFlag)
+	if err != nil {
+		return fmt.Errorf("invalid size: %w", err)
+	}
+
+	sizeGB := int(sizeBytes) / units.GB
+
+	if sizeFlag[0] == '+' {
+		volume, err := flapsClient.GetVolume(ctx, volID)
+		if err != nil {
+			return err
+		}
+		sizeGB += volume.SizeGb
+	}
+
 	if sizeGB == 0 {
```

Comment:
this check has to be right after line 82 and before checking for `+` char presence in `sizeFlag[0]`